### PR TITLE
Document how to configure proxy config

### PR DESCRIPTION
**Pull request by howardjohn**
How it looks rendered:

### CODE REVIEW

**Review by louiscryan — COMMENTED** [review]
LGTM

**Review by louiscryan — APPROVED** [review]

**Comment by kyessenov** [comment]
1. Why is file and environment variable method not mentioned? Is the plan to deprecate those sideloaded APIs?
2. Need explanation of how merging is done with both mesh and workload defined together.

**Comment by howardjohn** [comment]
> 1. Why is file and environment variable method not mentioned? Is the plan to deprecate those sideloaded APIs?
These are really, IMO, internal implementation details. There is no reason I know of that a user should ever touch them.

> 2. Need explanation of how merging is done with both mesh and workload defined together.
Agreed

**Comment by kyessenov** [comment]
1. That's fine, but we should work towards removing those features. It's weird to have a hidden way to bypass the API semantics defined here. It would be equally strange if some file overrides a virtual service.

**Comment by howardjohn** [comment]
>  It would be equally strange if some file overrides a virtual service.

You can configure this :slightly_smiling_face:

File: mesh/v1alpha1/proxy.proto [hunk 1]
```diff
@@ -291,7 +291,26 @@ message Topology {
...
+// This can also be configured on a per-workload basis by configuring the `proxy.istio.io/config` annotation on the pod. For example:
+//
+// ```
+// annotations:
+//   proxy.istio.io/config: |
+//     discoveryAddress: istiod:15012
+// ```
+//
+// If both are configured, the two are merged with per field semantics; and field set will fully replace that field.
+// This is different than a deep merge provided by protobuf or similar.
```
Comment (mandarjog) [inline mesh/v1alpha1/proxy.proto:313]: Clarify: annotation fields replace the defaultProxyConfig

**Comment by brian-avery** [comment]
This is awesome. Thanks @howardjohn. Spent too long trying to figure this out when working on https://istio.io/latest/docs/ops/configuration/telemetry/envoy-stats/

**Review by brian-avery — APPROVED** [review]

**Review by nrjpoddar — COMMENTED** [review]
@howardjohn should we mention that users need to restart workloads if they change at the mesh level?

**Pull request merged**